### PR TITLE
Increases HMS injection time

### PR DESCRIPTION
**Pull request by ghost**
## About The Pull Request

~~This increases the HMS autoinjector's use time to be a little longer (20s) than the eyesnatcher's use time (15s), in consideration of its impact.~~

This increases the HMS autoinjector's use time to be exactly the same as the eyesnatcher's use time (15).

## Todo

- [ ] Wait for https://github.com/tgstation/tgstation/pull/75654 's outcome - affects this code.
- [ ] Wait for https://github.com/tgstation/tgstation/pull/76291 - changes the balance of involved mechanics.

## Why It's Good For The Game

The eyesnatcher takes 15s to use; however, the eyesnatcher is far lower impact and its effects can be undone with but a short trip to medical.

In comparison, the HMS autoinjector leaves the threat of being dusted on a timer over the affected player for the entire round*, the mitigating medication only available from a single, fallible bottleneck.

\* There are some ways around this, but some are being removed even now and given it's intended to be incurable, I don't think we should balance around exploits liable to be fixed/closed.

In consideration of the much higher impact the HMS autoinjector has, I think it should not be so short many players won't react in time when not expecting confrontation - but rather, that it should take ~~a little longer than~~ at least as long as the eyesnatcher to use.

The rewards for completing these objectives are already the same.

## Some example scenarios

I think the gist is well enough above, but to better explain why I chose this time (and why small adjustments would not be sufficient), I want to give some example scenarios:

1. A victim who doesn't feel threatened is chatting (say, at the bar or medical or anywhere else), and - their focus being others they are interacting with or typing, they do not notice the injector line in time. I think this is highly problematic for something so impactful, and so I wish to close the door on it. Say, a 0.5s change would still allow it too much, in my opinion.

2. A victim does feel threatened, and the antagonist manages a hit from a stun prod or baton, and then a second followup hit once they're on the ground from the first hit. I'd call this problematic - less so than scenario one; some effort was put in here, but still too little considering what we're doing to them here. So this is another one I want to close out.

3. Consider a similar situation as scenario 2, but the antagonist restrains the victim (perhaps with sedatives or simple maint-fu instead of stun weaponry) and ferries them to maintenance or a department backroom or so on - though if they've attracted suspicion, someone may come to investigate in time. This is essentially what's required for the eyesnatcher, and I feel this level of effort is fair. Given the severity of HMS, I feel "around here" is the appropriate tuning for the objective.

## Changelog

:cl:
balance: Traitors' HMS autoinjectors now takes the same amount of time to use as an eyesnatcher.
/:cl:

### CODE REVIEW

File: code/modules/antagonists/traitor/objectives/infect.dm [hunk 1]
```diff
@@ -158,7 +158,7 @@
 /obj/item/reagent_containers/hypospray/medipen/manifoldinjector/attack(mob/living/affected_mob, mob/living/carbon/human/user)
 	if(used == 0)
 		to_chat(affected_mob, span_warning("You feel someone try to inject you with something."))
-		if(do_after(user, 1.5 SECONDS))
+		if(do_after(user, 20 SECONDS))
```
Comment (Mothblocks) [inline code/modules/antagonists/traitor/objectives/infect.dm:161]: Did you mean to 13x this value?

```suggestion
		if(do_after(user, 2 SECONDS))
```

Comment (YehnBeep) [reply code/modules/antagonists/traitor/objectives/infect.dm:161]: Yes.

The reference point is not the original value, but rather, other traitor objectives. The eyesnatcher seemed comparable since it's doing bad medical things to a character.

Another way to look at it - should this objective be 10x faster than the other objective? I don't think it should.

**Comment by SyncIt21** [comment]
Speaking of HMS, it might be good to merge #75654 since that PR mentions something about `do_after()` and them come back to this one

**Comment by YehnBeep** [comment]
> Speaking of HMS, it might be good to merge #75654 since that PR mentions something about `do_after()` and them come back to this

That's fair. Set to draft pending the other PR's outcome.

File: code/modules/antagonists/traitor/objectives/infect.dm (hunk 1 above)
Comment (YehnBeep) [reply code/modules/antagonists/traitor/objectives/infect.dm:161]: I've added some addition reasoning to the PR body on why I think this objective is tuned to the wrong order of magnitude. Though I don't expect this to exactly be super popular, I hope that can at least explain that this isn't a simple "kneejerk want to nerf by a huge amount" and rather to make it similar in effort and risks to lower impact objectives.

**Comment by carlarctg** [comment]
This is insane. Traitors already get comically, insultingly hard objectives that are not worth it on top of giving you barely any rewards.

Let's dissect this, point by point.

> The eyesnatcher takes 15s to use; however, the eyesnatcher is far lower impact and its effects can be undone with but a short trip to medical.

The eyesnatcher's impact is far HIGHER because it literally removes your ability to see until you go out of your way to have someone else help you via surgery, which isn't always a possibility. If you were infected, you can just do these things by yourself. Walk to medbay to get sans pills, or buy them from cargo - they're a goodie as well, as far as I know.

> In comparison, the HMS autoinjector leaves the threat of being dusted on a timer over the affected player for the entire round, the mitigating medication only available from a single, fallible bottleneck.

The 'threat of being dusted' is something that only happens if you ignore this for twenty actual minutes. I think that's pretty damn fair, certainly gives you a LOT of time to find the sans pills, which will leave you set for the rest of the round.

Here's the 'INSANE IMPACT' this disease has on you. You get infected, you notice, you grab a pill bottle from either cargo or medbay. You put it in your emergency box. You keep on going on the round, taking a pill every 20 minutes. 

We don't want traitors to be nothing but murder and boning, right? This is a step backwards, a pointless one, just because you're annoyed you got a slightly debilitating, mostly incurable disease (body transfer)

> In consideration of the much higher impact the HMS autoinjector has, I think it should not be so short many players won't react in time when not expecting confrontation - but rather, that it should take a little longer than the eyesnatcher to use.

The infection objective is actually interesting and unique in that it needs you to sabotage someone WITHOUT having to *actually* kill them. Changing this to a TWENTY SECOND windup (???) Just makes it another 'Kill them' objective like everything else. It's a terrible, terrible idea. ESPECIALLY with no improvements to the rewards for it.

Like, I legitimately see no reason that a traitor wouldn't just infect someone with this after they're dead, in maintenance, inside a locker. That's not an infection objective. That's a killing objective, and the infection thing is just some jank garbo you have to slap on at the end while standing still for about as long as it takes to draw a heretic rune. With the reward being one TC.

This is one of the only objectives you can reasonably accomplish without having a massive chance to blow your cover, and it's baffling that this is somehow a bad thing. When am I going to be able to stand still next to someone with an obvious dark epipen sprite in my hand to inject them?

It adds an interesting, unique scenario, with unique challenges - hang around someone for a bit, chat with them, act busy, and inject them with the HMS injector when they're preocuppied. They get a interesting disease that is not even debilitating in the slightest with a large, large safety net that adds some potential interaction with cargo and medbay. You're proposing changing this to something that is just, yet another 'Kill this dude, drag to maint, do whatever snowflake flavor your objective wants'

> I think the gist is well enough above, but to better explain why I chose this time (and why small adjustments would not be sufficient), I want to give some example scenarios:

>      A victim who doesn't feel threatened is chatting (say, at the bar or medical or anywhere else), and - their focus being others they are interacting with or typing, they do not notice the injector line in time. I think this is highly problematic for something so impactful, and so I wish to close the door on it. Say, a 0.5s change would still allow it too much, in my opinion.

Is this a bad thing? The traitor sabotaging someone without openly grabbing their .357 revolver?

>     A victim does feel threatened, and the antagonist manages a hit from a stun prod or baton, and then a second followup hit once they're on the ground from the first hit. I'd call this problematic - less so than scenario one; some effort was put in here, but still too little considering what we're doing to them here. So this is another one I want to close out.

There is almost no difference from this to actually capturing them. You're doing the exact same steps as that.

>     Consider a similar situation as scenario 2, but the antagonist restrains the victim (perhaps with sedatives or simple maint-fu instead of stun weaponry) and ferries them to maintenance or a department backroom or so on - though if they've attracted suspicion, someone may come to investigate in time. This is essentially what's required for the eyesnatcher, and I feel this level of effort is fair. Given the severity of HMS, I feel "around here" is the appropriate tuning for the objective.

No, it's not, and I'm baffled you think so. 

TL;DR: You shouldn't need to be batonged every single time something bad happens to you. You can get appendicitis, spontaneous brain trauma, diseases. Traitors shouldn't lose this unique objective that has interesting and unique ways of being  accomplished ingame, and doesn't severely impact the affected's round in doing so. This PR fixes a non-issue by adding frustration to traitors' objectives system, which needs the exact OPPOSITE of this, desperately so.

**Comment by YehnBeep** [comment]
> Is this a bad thing? The traitor sabotaging someone without openly grabbing their .357 revolver?

The issue is applying a very detrimental, irremovable mechanic to someone without a real chance at counterplay.

> They get a interesting disease that is not even debilitating in the slightest

You... don't know anything about HMS, do you? It's very debilitating. Vision disruptions, stuns, etc. So is taking the medication. It gets you coming and going.

----

Aside from that... There's really only two points I want to make, which are:

1. The blindness (and fracture and trauma) from the eyesnatcher are generally temporary. They're fixable and not tied to a single path to do so.
2. From what I've seen, plenty of people do eye snatching without first killing the victim.

When an antag wants to do something on this level to someone, I think it's fair to ask for similar effort to comparable objectives and for it to not be relatively free.

**Comment by Vect0r2** [comment]
Just make it curable and take like 2 seconds. That will make it annoying, but also let tots not have to murder for the objective.

**Comment by YehnBeep** [comment]
> Just make it curable and take like 2 seconds. That will make it annoying, but also let tots not have to murder for the objective.

I've been sitting on this for some time now thinking about how I might approach it.

The trouble is the original author/PR wanted to provide a very hardcore alternative to Brain Tumor. And they've sure done it.

But unlike Brain Tumor, it's not opt-in only. I do think having it be curable, or at least having alternative paths to obtain the medication, would make it more suitable to being inflicted on the general population. However... this would also hurt the experience for people that DO opt into it and want this very hardcore thing.

The advantage to this approach is it becomes a higher effort objective so if someone wants to alter another's round so much they have to work for it more, but also preserves the experience for people that take the quirk and like it as it is.

Also, honestly, I don't have particularly high hopes that this PR will go through and I think a more direct nerf to the disease would be even less likely to. If you think you can get support for it, though, it could be worth a try.

**Comment by YehnBeep** [comment]
Reduced the time to 15 seconds. Its use time and rewards now have exact parity with the eyesnatcher.

**Comment by ATH1909** [comment]
> The eyesnatcher takes 15s to use; however, the eyesnatcher is far lower impact and its effects can be undone with but a short trip to medical.

wouldn't the logical response thus be to shorten the eyesnatcher's application time? 15 seconds sounds absurd to me.

**Comment by YehnBeep** [comment]
> > The eyesnatcher takes 15s to use; however, the eyesnatcher is far lower impact and its effects can be undone with but a short trip to medical.
> 
> wouldn't the logical response thus be to shorten the eyesnatcher's application time? 15 seconds sounds absurd to me.

Well, it is a possible approach.

Would you be OK with the eyesnatcher being applied to people that have been stun baton'd but otherwise not restrained/disabled? If so, go ahead and shorten it to a few seconds.

Would you be OK with the eyesnatcher being applied as a driveby thing on people that are typing or otherwise at least minorly distracted? If so, then lower the time to the same as this injector's current value.

I don't think these would be good for the game, but that's always a bit subjective, and some would love easier objectives.

**Comment by ATH1909** [comment]
> I don't think these would be good for the game, but that's always a bit subjective, and some would love easier objectives.

I think the injector is SUPPOSED to be a driveby thing, yeah? Otherwise it's just stun->cuff/kill, same as a murder or kidnapping objective.

**Comment by YehnBeep** [comment]
> > I don't think these would be good for the game, but that's always a bit subjective, and some would love easier objectives.
> 
> I think the injector is SUPPOSED to be a driveby thing, yeah? Otherwise it's just stun->cuff/kill, same as a murder or kidnapping objective.

Well. And this brings us back to my reason for the PR. I do not think it is good for the game/players generally for a mechanic this high impact to be applied via a low effort objective. I've made my best arguments for why I think that already, so, not much more I can add just now.

**Comment by zxaber** [comment]
>But unlike Brain Tumor, it's not opt-in only. I do think having it be curable, or at least having alternative paths to obtain the medication, would make it more suitable to being inflicted on the general population. However... this would also hurt the experience for people that DO opt into it and want this very hardcore thing.

Sounds like the underlying problem is that a negative trait is being used as a traitor item. The drive-by inject thing would be fine if this just gave you a version of HMS that's curable by like intentionally gaining an addiction to a (reasonably complex) chem. The victim trades one downside for another, with a potential path to being cured of both eventually.

**Comment by Vect0r2** [comment]
> 

Nono, make the quirk stick. Make this apply a special removable HMS, not the normal quirk one.

**Comment by YehnBeep** [comment]
Hmm. Yeah, it does seem like a lot of the criticism to this PR isn't so much about the payload but that people like the objective's gameplay as is.

I might possibly be able to figure out the code enough to to do an HMS-lite that can be cured via enough of something to get an addiction. Um. I guess at this point, I could use maintainer input: Would this alternative be more (or less) appealing than changing the use time?

**Comment by newfren** [comment]
it is insane that it's currently 1 second

**Comment by MercuryArrow** [comment]
I think there's room for compromise here personally - 15 to 20 seconds feels much too long for the intent of the objective, but the existing 1-2 seconds is a little on the fast side.  I also agree with the sentiment that the eyesnatcher is worse due to its immediately debilitating impact.  HMS is worse in *theory* but has several different ways of treating it(whether this is intended or otherwise, it's a reality that stems from existing game systems and we should approach changes with that in mind) and doesn't become debilitating for a good 15 to 20 minutes after application - and even then, HMS can roll 'no effect' or something minor and reset itself back to stage 1.  The dusting and heart vanishment, while brutal, are not the only things it can do when it rolls over.

Honestly, I think the injector isn't too bad where it currently is due to the ease with which HMS can be managed even without cargo, particularly from mid-round onward. I'd say 5 seconds is more than enough time for the injection, anything beyond 15 is overkill.

**Comment by YehnBeep** [comment]
> Honestly, I think the injector isn't too bad where it currently is due to the ease with which HMS can be managed even without cargo, particularly from mid-round onward.

Can you please elaborate on what alternate methods exist?

**Comment by MercuryArrow** [comment]
> > Honestly, I think the injector isn't too bad where it currently is due to the ease with which HMS can be managed even without cargo, particularly from mid-round onward.
> 
> Can you please elaborate on what alternate methods exist?

Any form of aheal deals with it, as has been the bane of most negative traits and status effects since time immemorial - this means implanted legion cores and xenobio purple crossbreeds, primarily.  Various methods of body-swapping can fix it, particularly botanists doing podcloning.  Viral bonding surgery can lock HMS at stage 1, effectively curing it as it can't infect people.  And of course, you have the intended loop of doing a bounty a few times a shift to pay for the sansufentanyl pills themselves, and even that can be stretched out much further by breaking the pills down into slightly smaller doses or using other methods of ingestion.

I've personally seen all of these used over the course of the past week or two on Manuel, and that's really what puts me in the 'the objective as a whole is not a huge deal' camp - when even people on the MRP server are finding comfortable ways to get on with their round with a little creativity, I doubt the more fast paced sections of the community will have any trouble either.

**Comment by YehnBeep** [comment]
It's very, very rare that I see anyone doing Xenobio and willing to share the fruits thereof - also a pain since that's one of the very few ways to deal with deep rooted and perma traumas, but again, it's vanishingly rare to get someone willing to help a patient. Pod cloning I suppose tends to be bit more accessible with a good botanist.

That a surgery can lock it at stage 1... that is kind of a big deal though. Advanced surgeries are usually researched in my experience, then it's generally easy enough to ask for them. And then total negation for free or near enough.

**Comment by Exester509** [comment]
> > > I don't think these would be good for the game, but that's always a bit subjective, and some would love easier objectives.
> > 
> > 
> > I think the injector is SUPPOSED to be a driveby thing, yeah? Otherwise it's just stun->cuff/kill, same as a murder or kidnapping objective.
> 
> Well. And this brings us back to my reason for the PR. I do not think it is good for the game/players generally for a mechanic this high impact to be applied via a low effort objective. I've made my best arguments for why I think that already, so, not much more I can add just now.
> 
> As Vect0r2 mentioned - if one agrees there's a problem but not with this approach, another approach could be to leave the objective exactly as is and weaken HMS. But such an approach then hurts the integrity of the associated quirk and the original author's vision of HMS being a more extreme alternative to Brain Tumor.

It's literally a spammed objective that floods the tab for a good half of the round. Also it's absolutely meaningless and can be shrugged off with the slightest semblance of effort, I've only seen a single person get dusted and because they didn't even know what it was.
I guess murderboning IS the proper way to play traitor, if the only "stealthy" objective to do to a person is now put on a 15 second actionbar

**Comment by Bibarra** [comment]
God forbid traitors actually do something scary. God forbid traitor objectives have permanent round-long consequences and aren't nothingburgers. (And the consequences aren't even that bad in this case) God forbid there's tension and paranoia from someone standing next to you.

I liked how my character suddenly got dizzy and the realization as I scrolled up my chat box to see the "you feel a tiny prick".
With or without neat tricks to "cure" HMS, it impacts the round in interesting and unique ways that are not immensely detrimental to the player. I see there's a pr to prevent aheals and viral bonding from working, but even in that case getting the medicine is easy, bodyswapping and race changes are simple enough and have multiple ways of being performed, and of course there's also the chance that you might just die and that's not a bad thing.

A long timer on using the injector will encourage players to kill their target, then infect them. Stuns can't last long enough, and cuffing their target just to have enough time to inject them with an autoinjector when a syringe is so much faster is plain silly. Where's the stealth? The tension? The worry that someone standing next to me has an autoinjector in their pocket? I want MORE paranoia, not simply going about my business as usual because the injector time is so long that I'll never be stationary next to someone long enough.

**Comment by YehnBeep** [comment]
If people can do the other objective stealthily despite its 15 second time (and they do), I really doubt this is going to kill stealth traitoring.

It would kill driveby's, which is kind of the point of this PR: I don't think it's good balance to have something like HMS just applied because someone looked away from the log for all of a 1.5 or 2 seconds.

> can be shrugged off with the slightest semblance of effort

The most impactful thing I've seen that affects this is viral bonding, but that's apparently going away. HMS is not a 'shrugged off' kind of sickness.

**Comment by Exester509** [comment]
> If people can do the other objective stealthily despite its 15 second time (and they do), I really doubt this is going to kill stealth traitoring.
> 
> It would kill driveby's, which is kind of the point of this PR: I don't think it's good balance to have something like HMS just applied because someone looked away from the log for all of a 1.5 or 2 seconds.
> 
> > can be shrugged off with the slightest semblance of effort
> 
> The most impactful/practical thing I've seen that affects this is viral bonding, but that's very likely going away so. HMS is not a 'shrugged off' kind of sickness.

Oh really, they do? Only times when I've seen traitors use the eye snatcher stealthily and without killing the target is in very low pop and with stun weapons. 15 seconds is absurdly long for an objective that requires both people to stand still.

And yes, it can be shrugged off, chug a pill every now and then and you're golden, it doesn't progress fast enough for it to be a threat, and pills can be obtained fairly easily, not to mention plasmamen are already immune to the disease. Taking the eyes out is more debilitating and harmful than injecting HMS. I really hope this PR doesn't get merged, traitors aren't really in need of a nerf as big as this

**Comment by YehnBeep** [comment]
> Taking the eyes out is more debilitating and harmful than injecting HMS

Have to disagree on this point in particular. It's very easy to get new eyes - organ manip, don't need any research, can be done with starting materials.

Using the pills is debilitating. As is allowing the disease to progress to a point to stretch those pills. (Otherwise you're looking at an average of ~ 6 minutes which is not that long).

And um... did you just imply playing plasmamen is a solution to a balance problem..? (Though making the objective unable to roll for plasmamen targets may be worth a bugfix)

**Comment by Bibarra** [comment]
Monkify + Mutadone
Podclone
Bodyswap
Race change (xenobio or botany)
Just... Going to cargo and getting the damn pills yourself. (they aren't even expensive and I've never had difficulty obtaining them)
None of these options are difficult to obtain nor are being removed.

Just another local ided salt pr maker. If you're really going to go through with it despite being repeatedly shown to be wrong, at least reduce the weight of the objective so it doesn't flood my choices. (sometimes with duplicates of the same objective) Of course, the better change would be to make it and the eye snatcher 2 seconds, so a shovestun will do. That'd be scary and fun.

**Comment by YehnBeep** [comment]
> Just another local ided salt pr maker.

Here's a funny thing. I've never had HMS on my characters. Given I play silicons more than anything else, my characters are largely immune. But I do so the effects they have on others a lot, which was what ultimately led to me feeling strongly enough about this to make a PR.

As to your other solutions, many of them are being closed. The others might be got in time, except perhaps bodyswaps, but I don't consider that trivial.

But on the subject of salt, you don't seem to be here to be constructive, so I likely won't be responding to more of your comments unless that changes.

**Comment by Bibarra** [comment]
I apologize for expressing my dissatisfaction with this pr in that way.

I've seen my share of players infected with HMS and have been hit with it myself more than once. Personally I have never had any difficulty in dealing with it on either slow or chaotic rounds. I want to restate that the cures I listed in the last post are not removed by #76291 OR #75654. There will still be multiple viable ways to deal with HMS without using Viral Bonding or "aheal" type cures.

I'm not even opposed to some kind of progress time, but I would prefer if it were fair to the antagonist that would have to go out of their way to prepare for a 15 second ordeal. As I see it, kill and infect would be preferred and frankly we need less excuses for traitors to get their murder on.

**Comment by YehnBeep** [comment]
The common thread among these lines of criticisms is less concern about the payload and more that people like there being a quick, driveby'able objective. My issue with the balance of this is that I think the payload is too impactful to be so easily dispensed.

One previously suggested alternative is to have an HMS-like that can be dealt with and where such methods won't, either imminently by an existing PR or in a future bugfix, be closed - thus preserving the objective as is. This might be something I could do and I would be willing to try, but I also worry maintainers might not like there being two variants of HMS or other technical implications thereof.

I haven't received feedback on that yet, though I hope to. But if it's something that'd have a reasonable/good chance, I feel like that could fix the issues I (and maybe others) have with balance here while still preserving the gameplay people like of the objective.

But, supposing a worst case scenario for alternatives too: This PR is proving moderately unpopular. It's very possible the verdict will be a simple "no, it's fine" or "we'll allow a 0.5s increase only" - this latter thing having already been suggested by one of the more influential. I do feel like I have to try one thing or another, given what I've observed with this mechanic. But also - in your shoes - given the amount of feedback in favor of leaving the objective unchanged, I wouldn't lose hope on that.

**Comment by Jackal-boop** [comment]
a 15 second injection time for 1 TC is absolulty terrible. if you wanna really do this then increase the TC payout and make it a bit mor rare

**Comment by YehnBeep** [comment]
> a 15 second injection time for 1 TC is absolulty terrible. if you wanna really do this then increase the TC payout and make it a bit mor rare

Can someone point out to me what controls the rarity? I can't seem to discern the affective code for that.

**Comment by YehnBeep** [comment]
I need a break from this community, so I'm going to close this for now, and might as well wait for the other PRs to resolve anyways. I'll look into reopening when I'm back.

**Comment by RegJackson** [comment]
I never thought I'd see real humans feel so strongly about being entitled to free TC but this entire thread has made me firmly believe now more than ever that progtot was a mistake and we need to go the fuck back.

**Comment by Hardly3D** [comment]
Have we reached the point where traitor players are this entitled to want their objectives to be as easy as clicking on spaceman for free TC and rep? Even with the nerf, you guys do realize there are so many ways you can get away with injecting people without killing or going loud, right? (eg. radioactive microlaser, sleepy pen, chems, stun)

> This is insane. Traitors already get comically, insultingly hard objectives that are not worth it on top of giving you barely any rewards.

If you consider the current objectives "insultingly hard" then you seriously need to get good.

**Comment by YehnBeep** [comment]
The other PRs are done, but I'm too exhausted to pursue this further. If anyone would like to see a change like this, please consider opening your own PR.
